internal/services/backoff.go

```diff
@@ -57,9 +58,16 @@ func (b *Backoff) Delay() (time.Duration, error) {
 	if b.attempt == b.maxAttempts {
 		return 0, &MaxAttemptsExceeded{}
 	}
-	delay := time.Duration(b.scale * 1 << b.attempt)
-	if delay > b.max {
+	var delay time.Duration
+	// check that it doesn't overflow on int64
+	delayFloat := float64(b.scale) * math.Pow(2, float64(b.attempt))
```

Comment:
haven't you just swapped the possibility of an int64 overflow for an float64 one? :)

Reply:
ignore that, I'm wrong

Reply:
That's true but I was trying to be a little bit realistic :-)
Anyway let's back just to panic then.

Reply:
Approved, but to my taste, I prefer `if delay < 0 return 0, &MaxAttemptsExceeded{}" // guard overflow` rather than introduce the floating point complication.

Reply:
Happy about that. Going to change.

Reply:
@k-wall I finally simplified it but returning a more specific `BackoffDelayOverflow` error.